src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -163,6 +167,24 @@ def resolve(
         self.factory.preparer.prepare_linked_requirements_more(reqs)
         return req_set
 
+    def _prime_finder_cache(self, project_names: Iterable[str]) -> None:
+        """Populate finder's find_all_candidates cache
+
+        Pre-emptively call the finder's find_all_candidates for each project
+        in parallel in order to avoid later blocking on network requests during
+        resolution.
+        """
+
+        def _maybe_find_candidates(project_name: str) -> None:
+            try:
+                self._finder.find_all_candidates(project_name)
+            except AttributeError:
+                pass
```

Comment:
Could you add a comment that explains why it is necessary to catch `AttributeError`.

Reply:
I don't remember how it failed without it, and locally I can't currently get all the tests to pass on main. So, I could remove this again and see if ci catches it and if so add the comment, and if ci doesn't flag it then just let it be without error handling?

Reply:
Yes let's do that. We'll certainly learn something along the way.

Reply:
Removed the error-handler-wrapper in 14a6d3babfb50ebca2cd238cdb3f396f959de460 and ci passes. So I still don't know what the issue in the past was, which is a little frustrating, but now at least the code is a little simpler.

Reply:
Removed the error-handler-wrapper in 14a6d3babfb50ebca2cd238cdb3f396f959de460 and ci passes. So I still don't know what the issue in the past was, which is a little frustrating, but now at least the code is a little simpler.